src/sentry/integrations/discord/integration.py

```diff
@@ -39,7 +39,37 @@
 
 
 class DiscordIntegration(IntegrationInstallation):
-    pass
+    def get_client(self) -> DiscordClient:
+        org_integration_id = self.org_integration.id if self.org_integration else None
+
+        return DiscordClient(
+            integration_id=self.model.id,  # type:ignore
+            org_integration_id=org_integration_id,
+        )
+
+    def uninstall(self) -> None:
+        # If this is the only org using this Discord server, we should remove
+        # the bot from the server.
+        from sentry.services.hybrid_cloud.integration import integration_service
+
+        installations = integration_service.get_organization_integrations(
+            integration_id=self.model.id,
+            providers=["discord"],
+        )
+
+        # Remove any installations pending deletion
+        active_installations = [i for i in installations if i.status not in (2, 3)]
+
+        if len(active_installations) > 1:
+            return
+
+        client = self.get_client()
+        try:
+            client.leave_guild(str(self.model.external_id))
+        except ApiError:
+            # Most likely the bot has already been removed from the guild.
+            # In any case, we should just continue.
+            return
```

Comment:
Probably worth logging something in Sentry here - exit paths without logs/sentry errors make things tough to debug down the line